### DISCUSSION TITLE
技巧分享：通过 Umi-OCR 轻松实现截图翻译

### DISCUSSION THREAD

**Discussion by hiroi-sora** [crimx/ext-saladict#2051] [Show and tell]
# Umi-OCR 与 Saladict 搭配

<p align="center">
  <a href="https://github.com/hiroi-sora/Umi-OCR">
    <img width="200" height="128" src="https://tupian.li/images/2022/10/27/icon---256.png" alt="Umi-OCR">
  </a>
</p>

大家好，我是 [Umi-OCR](https://github.com/hiroi-sora/Umi-OCR) 的开发者。Umi-OCR 是一款开源离线OCR软件，使用简单，不需要注册账号或者申请API-key。有丰富的文本后处理功能，支持对各种段落布局进行分段合并。

在`v1.3.5`的更新中，Umi-OCR提供了与外部工具（如沙拉查词）联动的功能，可以对OCR识别出的文本进行翻译。

您无需编写脚本或进行复杂的操作，仅需在 Umi-OCR 中设置一组与 Saladict 相同的按键即可。

### 简要步骤（以Edge浏览器为例）：

1. 下载 [Umi-OCR](https://github.com/hiroi-sora/Umi-OCR)  ` v1.3.5` 及以上版本 。**勾选设置页底部的`高级选项`，重启软件。**
2. 打开沙拉查词的插件设置页面，左边栏选择`基本选项`，右边栏勾选`后台保持运行`。（如果不勾选也能使用，但必须保持浏览器开启。）
3. 左边栏点击`隐私设置`，右边栏点击`设置快捷键`。
    ![](https://tupian.li/images/2023/06/20/649165cae4b87.png)
4. 在弹出的新页面中，将沙拉查词的`在独立窗口中搜索剪贴板内容`设置任意一组快捷键，然后右边改为`全局`。
5. Umi-OCR勾选`自动复制结果`。然后`自动复制后发送按键`录制为相同快捷键，`1`次。（需要先勾选设置页底部的`高级选项`）

    ![](https://tupian.li/images/2023/06/20/649165faa434f.png)
6. 回到沙拉查词的设置页，左边栏点击`权限管理`，勾选`读取剪贴板`。
    ![](https://tupian.li/images/2023/06/20/649165f7dbf18.png)

7. 这样，Umi-OCR完成OCR后，会联动唤起沙拉查词进行翻译。你可以设置Umi-OCR为静默模式，使得每次任务只会弹出沙拉的窗口。

**Comment by hihvan** [comment]
啊，我正在想怎么联动这2个软件呢。发现作者在这里有分享。不过我在您的umi2.01中没有找到这个功能了。要是能够判断文本语言再选择是否发送按键可能会更方便一点 ，因为ocr中文文本大部分不需要翻译。
